test/e2e/pd.go

```diff
@@ -134,6 +135,66 @@ var _ = framework.KubeDescribe("Pod Disks", func() {
 		return
 	})
 
+	It("Same test as above, but remove pod with default grace period 30s [Slow]", func() {
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		By("creating PD")
+		diskName, err := createPDWithRetry()
+		framework.ExpectNoError(err, "Error creating PD")
+
+		host0Pod := testPDPod([]string{diskName}, host0Name, false /* readOnly */, 1 /* numContainers */)
+		host1Pod := testPDPod([]string{diskName}, host1Name, false /* readOnly */, 1 /* numContainers */)
+		containerName := "mycontainer"
+
+		defer func() {
+			// Teardown pods, PD. Ignore errors.
+			// Teardown should do nothing unless test failed.
+			By("cleaning up PD-RW test environment")
+			podClient.Delete(host0Pod.Name, api.NewDeleteOptions(0))
+			podClient.Delete(host1Pod.Name, api.NewDeleteOptions(0))
+			detachAndDeletePDs(diskName, []string{host0Name, host1Name})
+		}()
+
+		By("submitting host0Pod to kubernetes")
+		_, err = podClient.Create(host0Pod)
+		framework.ExpectNoError(err, fmt.Sprintf("Failed to create host0Pod: %v", err))
+
+		framework.ExpectNoError(f.WaitForPodRunningSlow(host0Pod.Name))
+
+		testFile := "/testpd1/tracker"
+		testFileContents := fmt.Sprintf("%v", mathrand.Int())
+
+		framework.ExpectNoError(f.WriteFileViaContainer(host0Pod.Name, containerName, testFile, testFileContents))
+		framework.Logf("Wrote value: %v", testFileContents)
+
+		// Verify that disk shows up for in node 1's VolumeInUse list
+		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, true /* shouldExist */))
+
+		By("deleting host0Pod")
+		//delete pod with default grace period 30s
+		framework.ExpectNoError(podClient.Delete(host0Pod.Name, &api.DeleteOptions{}), "Failed to delete host0Pod")
+
+		By("submitting host1Pod to kubernetes")
+		_, err = podClient.Create(host1Pod)
+		framework.ExpectNoError(err, "Failed to create host1Pod")
+
+		framework.ExpectNoError(f.WaitForPodRunningSlow(host1Pod.Name))
+
+		v, err := f.ReadFileViaContainer(host1Pod.Name, containerName, testFile)
+		framework.ExpectNoError(err)
+		framework.Logf("Read value: %v", v)
+
+		Expect(strings.TrimSpace(v)).To(Equal(strings.TrimSpace(testFileContents)))
+
+		// Verify that disk is removed from node 1's VolumeInUse list
+		framework.ExpectNoError(waitForPDInVolumesInUse(nodeClient, diskName, host0Name, nodeStatusTimeout, false /* shouldExist */))
+
+		By("deleting host1Pod")
+		framework.ExpectNoError(podClient.Delete(host1Pod.Name, api.NewDeleteOptions(0)), "Failed to delete host1Pod")
```

Comment:
Not in this PR but after it is merged we should address #28183. There we can clean up these two tests in both clarity and for maintenance.